scrapy_zyte_api/_middlewares.py

```diff
@@ -99,15 +98,7 @@ def process_request(self, request, spider):
         if self._param_parser.parse(request) is None:
             return
 
-        downloader = self._crawler.engine.downloader
-        slot_id = downloader._get_slot_key(request, spider)
-        if not isinstance(slot_id, str) or not slot_id.startswith(self._slot_prefix):
-            slot_id = f"{self._slot_prefix}{slot_id}"
-            request.meta["download_slot"] = slot_id
-        _, slot = downloader._get_slot(request, spider)
-        slot.delay = 0
```

Comment:
So, setting slot in the download middleware is not needed anymore?

Reply:
Right, we should keep the code to support requests that come from downloader middlewares. For such requests the queue issue would persist, since nothing will be able to set their slot before they reach the scheduler, but they would not be affected by the AutoThrottle extension at least.